### PR TITLE
Cp7m10 validate views post smarty update

### PR DESCRIPTION
**Pull request by victorvassilev**

### CODE REVIEW

File: www/include/options/media/images/DB-Func.php [hunk 1]
```diff
@@ -527,7 +527,7 @@ function getListDirectory($filter = null)
     $query .= "ORDER BY dir_name";
     $list_dir = array();
     $dbresult = $pearDB->query($query);
-    while ($row = $dbresult->fetchRow(DB_FETCHMODE_ASSOC)) {
+    while ($row = $dbresult->fetchRow(PDO::FETCH_ASSOC)) {
```
Comment (vhr) [inline www/include/options/media/images/DB-Func.php]: CentreonDBStatement::fetchRow don't have any arguments

Comment (victorvassilev) [reply www/include/options/media/images/DB-Func.php]: fixed

File: www/include/monitoring/status/TopCounter/xml/statusCounter.php [hunk 2]
```diff
@@ -139,7 +139,7 @@
         ") ";
 }
 $DBRESULT = $obj->DBC->query($query_svc_status);
-$svc_stat = array_map("myDecode", $DBRESULT->fetchRow());
+$svc_stat = array_map("myDecodeToInteger", $DBRESULT->fetchRow());
```
Comment (sc979) [inline www/include/monitoring/status/TopCounter/xml/statusCounter.php]: please change fetchRow() to fetch()

File: www/include/monitoring/status/TopCounter/xml/statusCounter.php [hunk 3]
```diff
@@ -139,7 +139,7 @@
...
 $DBRESULT->free();
```
Comment (sc979) [inline www/include/monitoring/status/TopCounter/xml/statusCounter.php]: please change free() to closeCursor()

File: www/include/options/media/images/DB-Func.php [hunk 4]
```diff
@@ -526,7 +526,7 @@ function getListDirectory($filter = null)
     }
     $query .= "ORDER BY dir_name";
     $list_dir = array();
-    $dbresult = $pearDB->query($query);
+
     while ($row = $dbresult->fetchRow()) {
```
Comment (sc979) [inline www/include/options/media/images/DB-Func.php]: you deleted $dbresult above.

Comment (victorvassilev) [reply www/include/options/media/images/DB-Func.php]: thanks, post-merge error

File: www/include/monitoring/status/TopCounter/xml/statusCounter.php (hunk 3 above)
Comment (victorvassilev) [reply www/include/monitoring/status/TopCounter/xml/statusCounter.php]: resolved

File: www/include/monitoring/status/TopCounter/xml/statusCounter.php (hunk 2 above)
Comment (victorvassilev) [reply www/include/monitoring/status/TopCounter/xml/statusCounter.php]: resolved

**Pull request merged**